mapflow/dialogs/mosaic_dialog.py

```diff
@@ -0,0 +1,62 @@
+from PyQt5 import uic
+from PyQt5.QtWidgets import QWidget, QDialogButtonBox
+
+from ..schema.data_catalog import MosaicCreateSchema, MosaicUpdateSchema, MosaicReturnSchema
+from .dialogs import ui_path, plugin_icon
+
+class MosaicDialog(*uic.loadUiType(ui_path/'mosaic_dialog.ui')):
+    def __init__(self, parent: QWidget) -> None:
+        """A dialog for adding or editing mosaics"""
+        super().__init__(parent)
+        self.setupUi(self)
+        self.setWindowIcon(plugin_icon)
+        self.ok = self.buttonBox.button(QDialogButtonBox.Ok)
+
+        self.mosaicName.textChanged.connect(self.on_name_change)
+
+    def on_name_change(self):
+        if not self.mosaicName.text():
+            self.ok.setEnabled(False)
+            self.ok.setToolTip(self.tr("Mosaic name must not be empty!"))
+        else:
+            self.ok.setEnabled(True)
+            self.ok.setToolTip("")
+
+class CreateMosaicDialog(MosaicDialog):
+    def __init__(self, parent: QWidget):
+        super().__init__(parent)
+        self.ok.setEnabled(False)
+
+    def setup(self):
+        self.setWindowTitle("")
+        self.mosaicName.setText("")
+        self.mosaicTags.setText("")
+        self.exec()
+
+    def mosaic(self):
+        if not self.mosaicName:
+            raise AssertionError("Mosaic name must not be empty!")
+        return MosaicCreateSchema(name = self.mosaicName.text(),
+                                  tags = [self.mosaicTags.text()] or None)
+
+class UpdateMosaicDialog(MosaicDialog):
+    def __init__(self, parent: QWidget):
+        super().__init__(parent)
+        self.ok.setEnabled(True)
+
+    def setup(self, mosaic: MosaicReturnSchema):
+        if not mosaic:
+            raise TypeError("UpdateMosaicDialog requires a mosaic to update")
+        self.setWindowTitle(self.tr("Edit mosaic {}").format(mosaic.name))
+        self.mosaicName.setText(mosaic.name)
+        if mosaic.tags:
+            self.mosaicTags.setText(", ".join(mosaic.tags))
+        else:
+            self.mosaicTags.setText("")
+        self.exec()
+
+    def mosaic(self):
+        if not self.mosaicName:
+            raise AssertionError("Mosaic name must not be empty!")
+        return MosaicUpdateSchema(name = self.mosaicName.text(),
```

Comment:
Hey, tags must be a list of str, and you just put them as a single string! There are two ways:
- parse comma-separated string (but need to clarify it to the user and validate them)
- use a set of lineEdit-s, each one representing a separate tag, wi a widget to add and remove the lines

Reply:
Also "or None" here will never work, as first operand will be at least [""], which resolves to True, because it is not an empty list.
It should have been 
[self.mosaicTags.text()] if self.mosaicTags.text() else None
(but anyway we need to change the tags input)

Reply:
Made a list from a string `self.mosaicTags.text().split(", ") if self.mosaicTags.text() else None`, so tags are `tags_list if self.mosaicTags.text() else []`. And added a label that explains it to user:
![image](https://github.com/user-attachments/assets/143982d6-8a7c-4d4f-a5a4-0f87683f06a3)